src/main/java/org/skriptlang/skript/bukkit/loottables/elements/effects/EffGenerateLoot.java

```diff
@@ -0,0 +1,80 @@
+package org.skriptlang.skript.bukkit.loottables.elements.effects;
+
+import ch.njol.skript.Skript;
+import ch.njol.skript.doc.Description;
+import ch.njol.skript.doc.Examples;
+import ch.njol.skript.doc.Name;
+import ch.njol.skript.doc.Since;
+import ch.njol.skript.lang.Effect;
+import ch.njol.skript.lang.Expression;
+import ch.njol.skript.lang.SkriptParser.ParseResult;
+import ch.njol.skript.lang.SyntaxStringBuilder;
+import ch.njol.util.Kleenean;
+import org.bukkit.event.Event;
+import org.bukkit.inventory.Inventory;
+import org.bukkit.loot.LootContext;
+import org.bukkit.loot.LootTable;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.Random;
+import java.util.concurrent.ThreadLocalRandom;
+
+@Name("Generate Loot")
+@Description({
+	"Generates the loot in the specified inventories from a loot table using a loot context. " +
+		"Some loot tables will require some of these values whereas others may not.",
+	"Note that if the inventory is full, it will cause warnings in the console due to over-filling the inventory.",
+})
+@Examples("generate loot of loot table \"minecraft:chests/simple_dungeon\" using loot context at player in {_inventory}")
+@Since("INSERT VERSION")
+public class EffGenerateLoot extends Effect {
+
+	static {
+		Skript.registerEffect(EffGenerateLoot.class,
+			"generate loot (of|using) [[the] loot[ ]table] %loottable% (with|using) [[the] [loot] context] %lootcontext% in [[the] inventor(y|ies)] %inventories%"
```

Comment:
It might be nice to have an option to generate without specifying a context, and have skript just auto-fill a context at the origin of a world